### DISCUSSION TITLE
freeze parameters and training

### DISCUSSION THREAD

**Discussion by Onkarsus13** [coqui-ai/TTS#2051] [General Q&A]
Whenever I am trying to freeze the parameters of the whole model instead of just the speaker embedding layer while training I am getting this error, can someone please give me a solution on this?

RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

**Comment by Edresson** [comment]
You can't freeze whole the model parameters and try to train the model. If every model module is frozen will not be any gradients to propagate. In addition, training with all weights frozen no make sense because the model will learn nothing. Try to just freeze parts of the model :).

**Comment by barghavanii** [comment]
> Whenever I am trying to freeze the parameters of the whole model instead of just the speaker embedding layer while training I am getting this error, can someone please give me a solution on this?
> 
> RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

hello can you assist me how you freeze some layers and keep fine tuning on the rest ?
